packages/mediapipe-task-text/lib/src/tasks/text_classification/text_classification_executor.dart

```diff
@@ -0,0 +1,150 @@
+  @visibleForTesting
+  Pointer<Pointer<Char>>? get createErrorMessage => _createErrorMessage;
+
+  /// Pointer to a possible error message in C memory for classification, used
+  /// to verify correct disposal.
+  @visibleForTesting
+  Pointer<Pointer<Char>>? get classifyErrorMessage => _classifyErrorMessage;
+
+  /// Pointer to a possible error message in C memory for closing the classifer,
+  /// used to verify correct disposal.
+  @visibleForTesting
+  Pointer<Pointer<Char>>? get closeErrorMessage => _closeErrorMessage;
+
+  /// Sends a `String` value to MediaPipe for classification. Blocks the main
+  /// event loop for the duration of the task, so use this with caution.
+  TextClassifierResult classify(String text) {
+    _createClassifier();
+    _classifyString(text);
+    return _processResult();
+  }
+
+  /// Creates a TextClassifier in C memory if one is not already on hand.
+  void _createClassifier() {
+    if (isNotNullOrNullPointer(_textClassifierPointer)) return;
+
+    // Allocate and hydrate the configuration object
+    _optionsPtr = options.toStruct();
+    _createErrorMessage = calloc<Pointer<Char>>();
+    _textClassifierPointer = bindings.text_classifier_create(
+      _optionsPtr!,
+      _createErrorMessage!,
+    );
+    _handleErrorMessage(_createErrorMessage!);
+    _createErrorMessage = null;
+  }
+
+  // Performs the actual classification.
+  void _classifyString(String text) {
+    _resultsPtr = calloc<bindings.TextClassifierResult>();
+
+    // Actually classify the text
+    _log.fine('Classifying $text');
+    _classifyErrorMessage = calloc<Pointer<Char>>();
+    final classificationStatus = bindings.text_classifier_classify(
+      _textClassifierPointer!,
+      prepareString(text),
+      _resultsPtr!,
+      _classifyErrorMessage!,
+    );
+    _handleErrorMessage(_classifyErrorMessage!, classificationStatus);
+    _classifyErrorMessage = null;
+  }
+
+  /// Converts the result from classification into a pure-Dart
+  /// [TextClassificationResult] object.
+  TextClassifierResult _processResult() {
+    // Convert the results into pure-Dart objects and free all memory
+    final result = TextClassifierResult.structToDart(_resultsPtr!.ref);
+    _log.fine('Text classification result: $result');
+    bindings.text_classifier_close_result(_resultsPtr!);
+    _resultsPtr = null;
+    return result;
+  }
+
+  /// Throws an exception if the [errorMessage] is non-empty.
+  ///
+  /// Also frees the memory backing [errorMessage], meaning that calling this
+  /// method completely handles the potential error.
+  void _handleErrorMessage(Pointer<Pointer<Char>> errorMessage, [int? status]) {
+    if (errorMessage.isNotNullPointer && errorMessage[0].isNotNullPointer) {
+      final dartErrorMessage = toDartStrings(errorMessage, 1);
+      _log.severe('dartErrorMessage: $dartErrorMessage');
+      if (status == null) {
+        calloc.free(errorMessage[0]);
+        calloc.free(errorMessage);
```

Comment:
It looks like these need to be called regardless of whether `errorMessage.isNotNullPointer && errorMessage[0].isNotNullPointer`. These also get missed in the next else that throws then.